feat(settings): Add checkbox to settings to enable or disable component

developerrowan opened the pull request:
Added a checkbox to the layout settings to allow the user to disable the component if they'd like, preventing split information they may not want to be sent to TheRun from being published. Specifically, this condition is checked in `HandleSplit` and `HandleReset`.

Added a setting to complement this in the XML settings, "IsEnabled", which is bound to said checkbox.

![](https://i.ibb.co/X3CWk94/therunlayout.png)

developerrowan committed:
feat(settings): Add checkbox to enable or disable component

Adds a checkbox to allow the user to disable the component, which will prevent split information from being uploaded when not desired

Adds a setting to the XML settings, "IsEnabled", which is bound to the checkbox

developerrowan committed:
Change bad checkbox name to something better

therungg commented:
Yo, thanks for the PR!! Couple of remarks:

1) I was actually thinking of doing this on the site itself. This way, it can be displayed along with other user settings, so users don't have settings in 2 different places, which might be confusing. Not a huge deal, but it might be preferable. Let me know what you think.

2) I'd prefer if there were two buttons, one for the Live page, and one for uploading stats after reset. Some people might want to just not be displayed on the Live page, but still have their stats synced automatically. Same goes other way around.

3) I could do it myself, but some administration has to be done, including adding a new build. Concretely this means `update.LiveSplit.TheRun.xml` should be updated, `UI/Components/CollectorFactory.cs`'s version should be updated, and lastly the .dll in `Components/LiveSplit.TheRun.dll` should be updated to the new build. This will trigger the auto-updater, so that people get a popup if they want to update.

4) Small thing, but I'd like the texts for the settings to align vertically. So "Upload Key' should be vertically aligned with "Enabled" (to the left of the checkbox).

In conclusion, I'm *alright* with having these specific settings in here, but won't be able to accept yet in this current form. Hope this makes sense! Let me know what you think.

developerrowan committed:
Changed checkbox to two checkboxes, updated minor, resolved comments

developerrowan commented:
Changed it to two checkboxes per your comment, updated the versioning to 2.0 (to match semver, since this is a minor, non-critical, non-breaking change). Due to the TableLayoutPanel, the upload key input had to be moved slightly to accomodate the checkboxes:

![image](https://user-images.githubusercontent.com/22936904/209035354-16075469-d6fd-4fe5-8391-f50c5b49c318.png)

I've tested this as thoroughly as I am able to, and it appears to behave as expected in all configurations.
